src/security/Crypto.js

```diff
@@ -1,135 +1,183 @@
      *
-     * @param {object} json - object to sign
-     * @param {object} keys - keys to sign with
-     * @return {string} signature - signature
+     * @param {Object} json - object to sign
+     * @param {Object} keys - keys to sign with
+     * @return {string} signature
      */
-    static signJson(json, keys) {
-        return Crypto.sign(stringify(json), keys);
+    static async signJson(json, keys) {
+        return await Crypto.sign(stringify(json), keys);
     }
 
     /**
-     * Signs a string and returns the signature
+     * Signs a string and returns the signature.
      *
      * @param {string} message - message to sign
-     * @param {object} keys - keys to sign with
-     * @return {string} signature - signature
+     * @param {Object} keys - keys to sign with
+     * @return {string} signature
      */
-    static sign(message, keys) {
+    static async sign(message, keys) {
+        if (!BROWSER) {
+            const msg = Crypto.wrapBuffer(message);
+            return base64Url(nacl.sign.detached(msg, keys.privateKey));
+        }
         const msg = Crypto.wrapBuffer(message);
-        return base64Url(nacl.sign.detached(msg, keys.secretKey));
+        let importedPrivateKey = keys.privateKey;
+        if (!importedPrivateKey instanceof CryptoKey) {
+            importedPrivateKey = await crypto.subtle.importKey(
+                'pkcs8',
+                importedPrivateKey,
+                {
+                    name: 'ECDSA',
+                    namedCurve: 'P-256',
+                },
+                false,
+                ['sign']
+            );
+        }
+        const sig = await crypto.subtle.sign(
+            {
+                name: 'ECDSA',
+                hash: {name: 'SHA-256'},
+            },
+            importedPrivateKey,
+            msg
+        );
+        const derSig = Crypto._convertSigToDER(sig);
+        return base64Url(derSig);
     }
 
     /**
      * Helper function for crypto engine createSigner:
-     * returns a signer that uses a keypair.
+     * returns a signer that uses a key pair.
      *
-     * @param {Key} keypair - such as returned by Token.Crypto.generatekeys
-     * @return {Object} signer, as expected from a crypto engine createSigner
+     * @param {Object} keyPair - such as returned by Token.Crypto.generateKeys
+     * @return {Object} signer object
      */
-    static createSignerFromKeypair(keypair) {
+    static createSignerFromKeypair(keyPair) {
         return {
-            sign: (message) => {
-                return Crypto.sign(message, keypair);
+            sign: async (message) => {
+                return await Crypto.sign(message, keyPair);
             },
-            signJson: (json) => {
-                return Crypto.signJson(json, keypair);
+            signJson: async (json) => {
+                return await Crypto.signJson(json, keyPair);
             },
-            getKeyId: () => keypair.id,
+            getKeyId: () => keyPair.id
```

Comment:
comma? did this pass the linter?

Reply:
sdk-js universally dont have trailing comma, for some reason, so i kept it consistent